tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/StringQueryExpressionTest.java

```diff
@@ -172,6 +178,15 @@ public void testQueryExprWithLimitForStringResultV2() {
         Assert.assertEquals(returnValues.toString(), "Ranjan John ");
     }
 
+    // issue - #40012
+    // @Test(description = "Negative Query expr for String tests")
+    // public void testNegativeQueryExprForXML() {
+    //     int index = 0;
+    //     validateError(negativeResult, index++, "ambiguous type '[string:Char, string:Char]'", 46, 16);
+    //     validateError(negativeResult, index++, "ambiguous type '[string:Char, string:Char, string:Char]'", 48, 16);
+    //     Assert.assertEquals(negativeResult.getErrorCount(), index);
+    // }
+
     @AfterClass
     public void tearDown() {
         result = null;
```

Comment:
Need to set the `negativeResult` to `null` as well.
```suggestion
        result = null;
        negativeResult = null;
```